File: scm-java/src/main/scheme/primitives/PrimitiveWsP.java
package scheme.primitives;
import scheme.*;

public class PrimitiveWsP extends Primitive {
    @Override
    public String name() { return "ws?"; }

    @Override
    public String info() {
        return "Syntax: (ws? x)\n" +
               "Library: (scm net websocket)\n" +
               "Description: Returns #t if x is a WebSocket object.\n" +
               "Example:\n" +
               "  (ws? ws) => #t";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 1, 1);
        return Value.isNativeValue(arguments[0]) && Value.asNativeValue(arguments[0]).value instanceof SchemeWebSocket
            ? Value.T : Value.F;
    }
}

File: scm-csharp/scheme/primitives/PrimitiveWsP.cs
namespace scheme;

public class PrimitiveWsP : Primitive
{
    public override string Name() => "ws?";

    public override string Info() =>
        "Syntax: (ws? x)\n" +
        "Library: (scm net websocket)\n" +
        "Description: Returns #t if x is a WebSocket object.\n" +
        "Example:\n" +
        "  (ws? ws) => #t";

    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 1, 1);
        return Value.IsNativeValue(arguments[0]) && Value.AsNativeValue(arguments[0]).value is SchemeWebSocket
            ? Value.T : Value.F;
    }
}
